azure/functions/decorators/function_app.py

```diff
@@ -1571,6 +1575,152 @@ def decorator():
 
         return wrap
 
+    def mcp_tool(self):
+        """
+        Decorator to register an MCP tool function.
+
+        Automatically:
+        - Infers tool name from function name
+        - Extracts first line of docstring as description
+        - Extracts parameters and types for tool properties
+        - Handles MCPToolContext injection
+        """
+        @self._configure_function_builder
+        def decorator(fb: FunctionBuilder) -> FunctionBuilder:
+            target_func = fb._function.get_user_function()
+            sig = inspect.signature(target_func)
+
+            # Pull any explicitly declared MCP tool properties
+            explicit_properties = getattr(target_func, "__mcp_tool_properties__", {})
+
+            # Parse tool name and description from function signature
+            tool_name = target_func.__name__
+            description = (target_func.__doc__ or "").strip().split("\n")[0]
+
+            # Identify arguments that are already bound (bindings)
+            bound_param_names = {b.name for b in getattr(fb._function, "_bindings", [])}
+            skip_param_names = bound_param_names
+
+            # Build tool properties
+            tool_properties = []
+            for param_name, param in sig.parameters.items():
+                if param_name in skip_param_names:
+                    continue
+                param_type_hint = param.annotation if param.annotation != inspect.Parameter.empty else str  # noqa
+
+                if param_type_hint is MCPToolContext:
+                    continue
+
+                # Inferred defaults
+                is_required = check_is_required(param, param_type_hint)
+                is_array = check_is_array(param_type_hint)
```

Comment:
I know extension define as isArray and boolean. Consider declare the name with name asArray, that makes little better in terms of readability.